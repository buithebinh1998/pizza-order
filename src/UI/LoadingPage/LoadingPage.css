.loading-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 100%;
  justify-content: center;
}

.loading-text {
  font-size: 2rem;
  font-weight: bold;
  color: #033a56;
}

.loader {
  border: 1rem solid #f3f3f3;
  border-radius: 50%;
  border-top: 1rem solid #3498db;
  width: 5rem;
  height: 5rem;
  margin-bottom: 1rem;
  -webkit-animation: spin 2s linear infinite; /* Safari */
  animation: spin 2s linear infinite;
}

/* Safari */
@-webkit-keyframes spin {
  0% {
    -webkit-transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@media (max-width: 992px) {
  .loading-text {
    font-size: 1.5rem;
  }

  .loader {
    border: 1rem solid #f3f3f3;
    border-radius: 50%;
    border-top: 1rem solid #3498db;
    width: 4rem;
    height: 4rem;
    margin-bottom: 1rem;
  }
}

@media (max-width: 768px) {
  .loading-text {
    font-size: 1rem;
  }

  .loader {
    border: 0.8rem solid #f3f3f3;
    border-radius: 50%;
    border-top: 0.8rem solid #3498db;
    width: 3.5rem;
    height: 3.5rem;
    margin-bottom: 1rem;
  }
}

@media (max-width: 576px) {
  .loading-text {
    font-size: 0.9rem;
  }

  .loader {
    border: 0.6rem solid #f3f3f3;
    border-radius: 50%;
    border-top: 0.6rem solid #3498db;
    width: 3rem;
    height: 3rem;
    margin-bottom: 1rem;
  }
}

@media (max-width: 320px) {
  .loading-text {
    font-size: 0.8rem;
  }

  .loader {
    border: 0.6rem solid #f3f3f3;
    border-radius: 50%;
    border-top: 0.6rem solid #3498db;
    width: 3rem;
    height: 3rem;
    margin-bottom: 1rem;
  }
}
