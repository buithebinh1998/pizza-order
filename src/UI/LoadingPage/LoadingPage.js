import React from 'react'
import './LoadingPage.css'

const LoadingPage = () => {
    return(
        <div className="loading-page">
            <div className="loading-text">Loading...</div>
            <div className="loader"></div>
        </div>
    )
}

export default LoadingPage;