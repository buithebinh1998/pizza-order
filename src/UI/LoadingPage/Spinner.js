import React from "react";
import "./LoadingPage.css";

const Spinner = () => {
  return (
    <div className="loading-page">
      <div className="loader" style={{ borderTopColor: "purple" }}></div>
    </div>
  );
};

export default Spinner;
