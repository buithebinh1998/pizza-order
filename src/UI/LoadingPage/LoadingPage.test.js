import { testSnapshots } from "../../utils/test.util";

import LoadingPage from "./LoadingPage";

describe("<LoadingPage / >", () => {
  testSnapshots(LoadingPage, [
    {
      props: {},
      description: "default render"
    }
  ]);
});
