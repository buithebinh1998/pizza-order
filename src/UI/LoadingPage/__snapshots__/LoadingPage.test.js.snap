// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingPage / > snapshots default render 1`] = `
<div
  className="loading-page"
>
  <div
    className="loading-text"
  >
    Loading...
  </div>
  <div
    className="loader"
  />
</div>
`;
